cmake/dts.cmake

```diff
@@ -119,15 +114,30 @@ if(SUPPORTS_DTS)
     ${NOSYSDEF_CFLAG}
     -D__DTS__
     -P
-    -E ${ZEPHYR_BASE}/misc/empty_file.c
+    -E   # Stop after preprocessing
+    -MD  # Generate a dependency file as a side-effect
     -o ${BOARD}.dts.pre.tmp
+    ${ZEPHYR_BASE}/misc/empty_file.c
     WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
     RESULT_VARIABLE ret
     )
   if(NOT "${ret}" STREQUAL "0")
     message(FATAL_ERROR "command failed with return code: ${ret}")
   endif()
 
+  # Parse the generated dependency file to find the DT sources that
+  # were included and then add them to the list of files that trigger
+  # a re-run of CMake.
+  toolchain_parse_make_rule(
+    ${PROJECT_BINARY_DIR}/${BOARD}.dts.pre.d
+    include_files # Output parameter
+    )
+
+  set_property(DIRECTORY APPEND PROPERTY
+    CMAKE_CONFIGURE_DEPENDS
```

Comment:
Hi,
I dont find CMAKE_CONFIGURE_DEPENDS property in cmake documentation.
I find CONFIGURE_DEPENDS property in cmake documentation.
Are both of them different?
In which version of cmake CMAKE_CONFIGURE_DEPENDS was available?
Thanks.